File: problemas/iniciante/1985/1985.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

let input = "";
let lineCount = 0;

process.stdin.on('data', (chunk) => {
    input += chunk;
});

process.stdin.on('end', () => {
    const estoque = {1001: 1.50, 1002: 2.50, 1003: 3.50, 1004: 4.50, 1005: 5.50};

    let entrada = input.trim().split('\n');
    let N = parseInt(entrada.shift());
    let compras = [];
    let pagar = 0;
    for (let i = 0; i < N; ++i) {
        const [codigo, quantidade] = entrada.shift().trim().split(' ').map(Number);
        compras.push([codigo, quantidade]);
    }

    compras.forEach(pedido => {
        const [codigo, quantidade] = pedido;
        if  (estoque.hasOwnProperty(codigo)) {
            pagar += quantidade * estoque[codigo];
        }
    });

    console.log(pagar.toFixed(2));
});

File: problemas/iniciante/1985/1985.py
def main():
    estoque = {1001: 1.50, 1002: 2.50, 1003: 3.50, 1004: 4.50, 1005: 5.50}
    entrada = []
    pagar = 0
    p = int(input())
    for _ in range(p):
        codigo, quantidade = map(int, input().split())
        entrada.append([codigo, quantidade])
    
    for pedido in entrada:
        if pedido[0] in estoque:
            pagar += pedido[1] * estoque[pedido[0]]
    print(f"{pagar:.2f}")

if __name__ == "__main__":
    main()
